src/main/kotlin/no/skatteetaten/aurora/boober/service/RedeployService.kt

```diff
@@ -1,99 +1,74 @@
 package no.skatteetaten.aurora.boober.service
 
 import com.fasterxml.jackson.databind.JsonNode
-import com.fasterxml.jackson.databind.node.ArrayNode
 import no.skatteetaten.aurora.boober.model.AuroraDeploymentSpec
 import no.skatteetaten.aurora.boober.model.TemplateType
-import no.skatteetaten.aurora.boober.service.openshift.OpenShiftClient
-import no.skatteetaten.aurora.boober.service.openshift.OpenShiftResponse
-import no.skatteetaten.aurora.boober.utils.openshiftKind
-import no.skatteetaten.aurora.boober.utils.openshiftName
+import no.skatteetaten.aurora.boober.service.openshift.*
 import org.springframework.stereotype.Service
 
 @Service
 class RedeployService(val openShiftClient: OpenShiftClient, val openShiftObjectGenerator: OpenShiftObjectGenerator) {
 
-    data class ImageInformation(val lastTriggeredImage: String, val imageStreamName: String, val imageStreamTag: String)
+    data class RedeployResult @JvmOverloads constructor(
+            val success: Boolean = true,
+            val message: String? = null) {
 
-    fun triggerRedeploy(deploymentSpec: AuroraDeploymentSpec, openShiftResponses: List<OpenShiftResponse>): List<OpenShiftResponse> {
+        companion object {
+            fun fromOpenShiftStatus(openShiftStatus: OpenShiftStatus): RedeployResult {
+                val success = openShiftStatus.openShiftResponses.all { it.success }
+                val message = if (success) "Redeploy succeeded" else "Redeploy failed"
+                return RedeployResult(success = success, message = message)
+            }
+        }
+    }
 
-        val namespace = deploymentSpec.environment.namespace
+    fun triggerRedeploy(deploymentSpec: AuroraDeploymentSpec, status: OpenShiftStatus): RedeployResult {
 
-        val redeployResourceFromSpec = generateRedeployResourceFromSpec(deploymentSpec, openShiftResponses) ?: return emptyList()
+        val namespace = deploymentSpec.environment.namespace
+        val redeployResourceFromSpec = generateRedeployResourceFromSpec(deploymentSpec, status) ?: return RedeployResult()
```

Comment:
I feel that the result of this method is handled a little to general. It was that way before and I think there is room for improvement here. 

Basically if we do a DeploymentRequest here we can also just exit. Not only if we get a null. 

Should we perhaps refactor out the first guards of generateRedeployresourceFromSpec and move that out of this method? Then this method will only handle the case where we actually have an imageStream that we know we have to update?